course5/week2/ex02.cc

```diff
@@ -0,0 +1,89 @@
+#include <iostream>
+#include <cmath>
+#include <chrono>
+
+// ARS - 20170213
+// Cormacks algo from https://en.wikipedia.org/wiki/Fast_inverse_square_root
+// With slight modifications to C++-ifyusing dynamic c
+
+float Q_sqrt(float);
+
+float Q_sqrt( float number )
+{
+	long i;
+	float x2, y;
+	const float threehalfs = 1.5F;
+
+	x2 = number * 0.5F;
+	y  = number;
+
+	// ARS: Original was: i  = * ( long * ) &y;
+	i  = * reinterpret_cast<long *> (&y);                       // evil floating point bit level hacking
+	i  = 0x5f3759df - ( i >> 1 );               // what the fuck? 
+	// ARS. Original was: y  = * ( float * ) &i;
+	y  = * reinterpret_cast<float *> (&i);
+	y  = y * ( threehalfs - ( x2 * y * y ) );   // 1st iteration
+
+	return number*y;
+}
+
+void TimeIt(int max_nr)
+{
+    // OBSERVATIONS: Hack seems to be much slower than std:sqrt???
+    // Iterating over e.g. 1E06 numbers, gives 50ms for Cormacks, vs 7 for std::sqrt. Dos the latter use native assembly sqrt??
```

Comment:
Would native assembly be faster?
Have you compiled it in debug or in release mode?